### ISSUE TITLE
Use the system configured blink rate for the editor's cursor

### ISSUE THREAD

**Issue by PDeets** [microsoft/vscode#131941]
Issue Type: <b>Bug</b>

In the Windows start menu, type "cursor blink rate" to search for the cursor blink rate setting. Configure this setting to be "None" in order to disable text input cursor blinking.

Try out various places where you can input text in VS Code. Some honor the setting and some do not.

Here are examples of places that do not honor the setting:
* In the Git view, click the Message text input to write in a message. The cursor will blink here.
* Debug a process and click the command input in the "debug console". The cursor will blink here.

Here are examples of places that honor the setting:
* In the file view, click the "new file" icon to get a text input field where you 
can enter the filename. The cursor will not blink here.
* In the search view, click "search" and "replace" to enter text. The cursor will not blink here.
* In the terminal, enter a command. The cursor will not blink here.

The setting should be consistently honored.

(Note that this is an accessibility bug. It may not have a high hit rate because it isn't a commonly needed accessibility setting, but for people like me who depend on this setting, it is a very annoying bug. When the text input cursor blinks, it is extremely distracting to me and I have a very hard time focusing on what I am doing.)

VS Code version: Code 1.59.1 (3866c3553be8b268c8a7f8c0482c0c0177aa8bfa, 2021-08-19T11:56:46.957Z)
OS version: Windows_NT x64 10.0.22448
Restricted Mode: No

**Comment by AuracleTech** [comment]
https://user-images.githubusercontent.com/40531871/158274357-2e0764eb-e212-45ef-8fd8-c6f5ced587e5.mp4

**Comment by lszomoru** [comment]
@alexdima, @hediet, it seems to be that the editor does not honour the operating system's cursor blink rate setting. This means that all workbench inputs that are using the editor do not honour this setting (ex: git commit message, search field in the extensions viewlet, etc.). The search viewlet is using the vanilla input element which honours the setting.

Is this a known issue with the editor? Could you please investigate this? Thank you!

**Comment by alexdima** [comment]
I don't believe there is any browser API that would allow us to read the system caret blink time. On Windows, there is a [`GetCaretBlinkTime()`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getcaretblinktime) in the win32 API. After a search in Electron, I could find it used and eventually set [here](https://github.com/electron/electron/blob/108ee7037f09fd623f473af5b3ecd943b5f6014a/shell/browser/api/electron_api_web_contents.cc#L847-L849). @deepak1556 What do you think, would exposing the system caret blink time be a reasonable feature request for Electron?

**Comment by deepak1556** [comment]
@alexdima do we use a custom cursor implementation for the editor ? Runtime uses the api to honor the interval for text inputs, so it is reasonable to expose it as an api if it helps custom implementations to sync with those intervals.

**Comment by hediet** [comment]
> do we use a custom cursor implementation for the editor ?

Yes

**Comment by rzhao271** [comment]
I turned on editor accessibility support and tried various editor.cursorBlinking setting values, but the cursor still blinks in the Settings editor search box.

**Comment by alexdima** [comment]
@rzhao271 Good find! In https://github.com/microsoft/vscode/pull/168217 we adopt that the setting is forwarded in the repl and in the SCM input. Please feel free to use this issue to adopt forwarding this editor setting for the editor embedded in settings search!

**Comment by dbaeumer** [comment]
I will reopen the issue. Please close again if this works as expected.

**Comment by rzhao271** [comment]
@dbaeumer right now the blinking is controlled by the `editor.cursorBlinking` setting rather than the OS.
I fixed an issue where the Settings editor search inputbox wasn't picking up that setting properly, but I don't think the blinking ever corresponded to the OS.

Verification steps:
1. Set `editor.cursorBlinking` to a value such as `expand` or `solid` in the Settings UI.
2. Click on the settings search box. The cursor blinking behaviour should correspond to the set value in step 1.

**Comment by kryptan** [comment]
> right now the blinking is controlled by the editor.cursorBlinking setting rather than the OS.

I think correct behaviour would be to use the OS settings by default. I generally expect my apps to honour global OS settings.
